curl/src/main/scala/org/http4s/curl/http/RequestSend.scala

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2022 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s.curl.http
+
+import cats.effect._
+import cats.effect.implicits._
+import cats.effect.std._
+import fs2.Pipe
+import org.http4s.curl.internal.Utils
+import org.http4s.curl.unsafe.libcurl_const
+import scodec.bits.ByteVector
+
+import scalanative.unsafe._
+import scalanative.unsigned._
+
+final private[curl] class RequestSend private (
+    flowControl: FlowControl,
+    requestBodyChunk: Ref[SyncIO, Option[ByteVector]],
+    requestBodyQueue: Queue[IO, Unit],
+    sendPause: Ref[SyncIO, Boolean],
```

Comment:
Is this `sendPause` redundant to the one in `flowControl`?

Reply:
Oops! I missed it in the refactoring process, however it's strange why compiler didn't warn about unused values :thinking:

Reply:
Yes, it was using the wrong ref, which caused the compiler not to warn about it, just pushed a fixed version.